### PR TITLE
Add H36M dataset for 2D keypoint detection

### PR DESCRIPTION
**Pull request by ly015**
* add TopDownH36MDataset
* support human3.6m 2d keypoint  inference
* add experiment configs

### CODE REVIEW

File: configs/body/2d_kpt_sview_rgb_img/topdown_heatmap/h36m/hrnet_w32_h36m_256x256.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        pipeline=train_pipeline),
+    val=dict(
+        type='TopDownH36MDataset',
+        ann_file=f'{data_root}/annotation_body2d/h36m_coco_test.json',
+        img_prefix=f'{data_root}/images/',
+        data_cfg=data_cfg,
+        pipeline=val_pipeline),
+    test=dict(
+        type='TopDownH36MDataset',
+        ann_file=f'{data_root}/annotation_body2d/h36m_coco_test_debug.json',
```
Comment (jin-s13) [inline configs/body/2d_kpt_sview_rgb_img/topdown_heatmap/h36m/hrnet_w32_h36m_256x256.py:157]: rm debug

File: mmpose/datasets/datasets/top_down/topdown_h36m_dataset.py [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+                :heatmap (np.ndarray[N, K, H, W]): model output heatmap
+                :bbox_id (list(int)).
+            res_folder (str): Path of directory to save the results.
+            metric (str | list[str]): Metric to be performed. Defaults: 'mAP'.
+
+        Returns:
+            dict: Evaluation results for evaluation metric.
+        """
+        metrics = metric if isinstance(metric, list) else [metric]
+        allowed_metrics = ['EPE']
```
Comment (jin-s13) [inline mmpose/datasets/datasets/top_down/topdown_h36m_dataset.py:203]: Why not support more metrics?

Comment (ly015) [reply mmpose/datasets/datasets/top_down/topdown_h36m_dataset.py:203]: Human3.6M is not a commonly used dataset for 2d keypoint detection, and we add this config and model mainly for two-stage 3d keypoint demos. So maybe extensive evaluation is unnecessary here?

Comment (jin-s13) [reply mmpose/datasets/datasets/top_down/topdown_h36m_dataset.py:203]: I suggest using PCK or AP as the default metrics.

File: configs/body/2d_kpt_sview_rgb_img/topdown_heatmap/h36m/hrnet_w32_h36m_256x256.py (hunk 1 above)
Comment (ly015) [reply configs/body/2d_kpt_sview_rgb_img/topdown_heatmap/h36m/hrnet_w32_h36m_256x256.py]: fixed

Comment (ly015) [reply configs/body/2d_kpt_sview_rgb_img/topdown_heatmap/h36m/hrnet_w32_h36m_256x256.py]: fixed

File: mmpose/datasets/datasets/top_down/topdown_h36m_dataset.py (hunk 2 above)
Comment (ly015) [reply mmpose/datasets/datasets/top_down/topdown_h36m_dataset.py]: Updated to use PCK@0.05 as the default metrics

**Review by jin-s13 — APPROVED** [review]

**Pull request merged**